### ISSUE TITLE
[BUG]  `series-set-item` ignores index when setting with another `cudf-series`

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#3454]
**Describe the bug**
 `series-set-item` ignores index when setting with another `cudf-series`

**Steps/Code to reproduce bug**

```python
df = cudf.DataFrame(data={ "a": [1, 2, 3],"b": [-1, -2, -3]}, index=[1, 2, 3])
df["b"] = cudf.Series(data=[12, 11, 10], index=[3, 2, 1])
print(df)

  a   b
1  1  12
2  2  11
3  3  10
```

**Expected behavior**
```python
df = pd.DataFrame(data={ "a": [1, 2, 3],"b": [-1, -2, -3]}, index=[1, 2, 3])
df["b"] = pd.Series(data=[12, 11, 10], index=[3, 2, 1])
print(df)


   a   b
1  1  10
2  2  11
3  3  12
```

**Environment overview (please complete the following information)**
 - Environment location: 
 - Method of cuDF install: [conda]

```
cudf                      0.11.0a191125         py37_3390    rapidsai-nightly
libcudf                   0.11.0a191125     cuda10.0_3390    rapidsai-nightly
```

**Additional context**
This behavior is also present when setting item with `cudf-dataframe`. 

Relevant PR : https://github.com/rapidsai/cudf/pull/3442

**Comment by jakirkham** [comment]
Now that PR ( https://github.com/rapidsai/cudf/pull/3442 ) is in, should this be closed?

**Comment by kkraus14** [comment]
Resolved by #3799
